### PR TITLE
Made Preferences API consistent with others

### PR DESCRIPTION
**Pull request by Redth**
### Description of Change ###

Made Preferences API all statics to be more consistent with the other API's in the library.

### API Changes ###

All `Get`/`Set` methods are now static, and have overloads to take a `string sharedName` parameter.

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Has samples (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Updated documentation ([see walkthrough](https://github.com/xamarin/Caboodle/wiki/Documenting-your-code-with-mdoc))

### CODE REVIEW

**Comment by mattleibow** [comment]
This looks OK. Just a question about setting a value to `null` - what happens. The switch statement will not actually get called:

```csharp
Prefs.SetValue("key", "value");
// the `case string s:` is not valid as `s` is not actually a string - it is null
Prefs.SetValue("key", (string)null);

// val will == "value"
var val = Prefs.GetValue("key", "");
```

At least that is what I _think_ may happen. Do we have a test for this?

**Comment by jamesmontemagno** [comment]
I thought the idea was if someone passed in "null" then we would just remove that key?

**Comment by jamesmontemagno** [comment]
Should we pull this in @Redth ? any other updates based on @mattleibow's stuff?

**Comment by mattleibow** [comment]
This test fails:

```csharp

[Theory]
[InlineData("string1", "TEST", null)]
[InlineData("string1", "TEST", sharedNameTestData)]
public void Set_Set_Null_Get_String(string key, string value, string sharedName)
{
    Preferences.Set(key, value, sharedName);
    Preferences.Set(key, null, sharedName);

    Assert.Null(Preferences.Get(key, null, sharedName));
}
```

File: Caboodle/Preferences/Preferences.android.cs [hunk 1]
```diff
@@ -6,141 +6,152 @@
...
+                                if (string.IsNullOrWhiteSpace(savedDouble))
                                 {
-                                    var maxString = Convert.ToString(double.MaxValue, CultureInfo.InvariantCulture);
-                                    outDouble = savedDouble.Equals(maxString) ? double.MaxValue : double.MinValue;
+                                    value = defaultValue;
                                 }
+                                else
+                                {
+                                    double outDouble;
+                                    if (!double.TryParse(savedDouble, out outDouble))
```
[the remark this answers is not in the document]
Comment (jamesmontemagno) [reply Caboodle/Preferences/Preferences.android.cs]: Yes, i think this is wrong.

Should be:  if (!double.TryParse(savedDouble, NumberStyles.Number | NumberStyles.AllowExponent, CultureInfo.InvariantCulture, out double outDouble))
                        {

File: Caboodle/Preferences/Preferences.android.cs [hunk 2]
```diff
@@ -6,141 +6,151 @@
...
                 }
             }
         }
 
-        T Get<T>(string key, T defaultValue)
+        static T PlatformGet<T>(string key, T defaultValue, string sharedName)
         {
             lock (locker)
             {
+                var actualType = Type.GetTypeCode(typeof(T));
```
Comment (mattleibow) [inline Caboodle/Preferences/Preferences.android.cs]: Besides this, I think it is good.

**Pull request merged**